### PR TITLE
fix(compiler): bitwise NOT (~) was a silent identity (B-766)

### PR DESCRIPTION
**Pull request by antoniosarosi**
### Bug
The `~` prefix operator compiled and ran with **no diagnostic** but was an identity function (`~x == x`):

| expr | before | expected |
|------|--------|----------|
| `~0` | `0` | `-1` |
| `~1` | `1` | `-2` |
| `~255` | `255` | `-256` |
| `~(-1)` | `-1` | `0` |

The other bitwise ops (`& | ^ << >>`) were already correct.

### Root cause
`~` (`Tilde`) had **no match arm** in `lower_unary_expr` (`baml_compiler2_ast/src/lower_expr_body.rs`). The token fell through to `_ => {}`, so no `UnaryOp` was recorded and the function returned the operand expression unchanged — the operator was silently dropped during CST→AST lowering, before it ever reached the type checker, MIR, or VM. (So this was never a wrong opcode in the VM; the op simply vanished at lowering.)

### Fix
Desugar `~x` into `-x - 1` (two's-complement complement) in AST lowering, reusing the already-correct `Neg` and `Sub`. This:
- keeps the change entirely in the AST-lowering layer — **no TIR / type-system changes**, no new operator threaded through MIR/VM;
- evaluates the operand exactly once (no double side effects);
- is correct for `int` and `bigint`.

Verified via `baml-cli run -e`: `~0→-1`, `~1→-2`, `~255→-256`, `~(-1)→0`, `~(~5)→5`, `~INT_MAX→INT_MIN`. Non-int operands now give clean type errors (`~true`, `~"hi"`) instead of a silent identity.

### Known corner: `~INT_MIN`
`~INT_MIN` mathematically equals `INT_MAX` (which is representable), but the intermediate `-INT_MIN` overflows BAML's 63-bit range, so it errors — exactly like plain `-(INT_MIN)`, which already errors in BAML today. This is an inherent limitation of negating INT_MIN in a 63-bit signed system, **not** a silent wrong answer (which is the bug this removes). Making `~INT_MIN` fully correct would require a dedicated VM bit-complement opcode plus a TIR type rule for a distinct `~` operator, which is out of this ticket's scope (VM/operator correctness fix; TIR owned by another engineer).

### Regression test
`crates/baml_tests/baml_src/ns_operators/operators.baml` gains a `bitwise_not` test with the exact ticket vectors (`~0==-1`, `~1==-2`, `~255==-256`, `~(-1)==0`). Regenerated the `operators` bytecode snapshot — only that namespace shifted (new test lambda + renumbered later lambdas); no `__baml_std__` churn.

### Secondary (skipped) — `~` in `baml describe Int`
The `int.baml` builtin doc only documents arithmetic operators; there is no bitwise-operators section to append `~` to, and that file lives in `baml_builtins2/baml_std`, which regenerates `__baml_std__` snapshots (the churn / sibling-contamination zone flagged for this machine). Skipped per the ticket's "skip if it widens the change" guidance.

### Local CI (0 failures)
- `cargo nextest run --all-features -E 'not package(baml_tests) and not package(/^sdk_test_/) and not binary(=pack_e2e)'` → 4245 passed
- `cargo test -p baml_tests` → all binaries passed
- `cargo fmt --check` + `cargo clippy --all-targets --all-features -- -D warnings` → clean

## Summary by CodeRabbit

* **New Features**
  * Added support for the `~` bitwise NOT operator in expressions.

* **Bug Fixes**
  * `~` now follows standard two’s-complement behavior (e.g., `~0 = -1`, `~1 = -2`, `~255 = -256`, `~(-1) = 0`), including edge-case correctness.

* **Tests**
  * Added a dedicated `bitwise_not` test block covering multiple cases to prevent regressions.

### CODE REVIEW

File: baml_language/crates/baml_compiler2_ast/src/lower_expr_body.rs [hunk 1]
```diff
@@ -1332,6 +1338,35 @@ impl LoweringContext {
...
+            return self.alloc_expr(
+                Expr::Binary {
+                    op: BinaryOp::Sub,
+                    lhs: neg,
+                    rhs: one,
+                },
+                span,
+            );
+        }
+
```
[the remark this answers is not in the document]
Comment (antoniosarosi) [reply baml_language/crates/baml_compiler2_ast/src/lower_expr_body.rs:1379]: Fixed in 13de6378b. The `~x -> -x - 1` desugar now wraps its `alloc_expr` calls in the established `self.synthesizing` save/restore pattern (same as the backtick/tagged-template desugarings), so the generated `Neg`/`Sub`/literal nodes are recorded in `source_map.synthetic_exprs`. Since the whole `~x` desugars away (no surviving node maps 1:1 to source, unlike the `Template` wrapper), all three generated nodes are marked synthetic; only the operand keeps its real, non-synthetic id.
